### PR TITLE
Fix a bug where the NUnit-Report fails to generate if the test output contains Virtual Terminal Sequences

### PR DESCRIPTION
**Pull request by ocalvo**
## PR Summary

The following test will fail to generate an NUnit report:

```
Describe 'Describe VT Sequences' {
  Write-Output "`e[32mHello World`e[0m"
  $true | Should -Be $true
}
```

This will fail because WriteCData does not support writting the ESC character, which is a valid [VT sequence](https://learn.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences).
The fix is to replace the ESC and Bell characters for {ESC} and {BELL}. 
According to the [Xml CDATA spec](https://www.w3.org/TR/REC-xml/#NT-Char), there is no way to escape these characters, other than failing, but we still need to represent them in the NUnit report.
Afte this change, these characters will be represented as by its printable version in the unicode table: `␛` for the `ESC` (0x1B) and as `␇` for the `BELL` (0x07) in the resulting Xml CData for example.

## PR Checklist

- [x] PR has meaningful title
- [x] Summary describes changes
- [x] PR is ready to be merged
  - If not, use the arrow next to `Create Pull Request` to mark it as a draft. PR can be marked `Ready for review` when it's ready.
- [x] Tests are added/update *(if required)*
- [ ] Documentation is updated/added *(if required)*

### CODE REVIEW

**Comment by nohwnd** [comment]
Hello, thanks for the PR, the test you added is failing. I assume you also want this backported to version 5.x.x ?

**Comment by ocalvo** [comment]
> Hello, thanks for the PR, the test you added is failing. I assume you also want this backported to version 5.x.x ?

No, we can easily update our test suite to latest version.

**Comment by ocalvo** [comment]
@nohwnd I have fixed the test, it was failling because WindowsPowerShell does not escape the ESC character. Fixed using the ASCII value explicitly.

File: src/functions/TestResults.NUnit3.ps1 [hunk 1]
```diff
@@ -537,11 +537,27 @@ function Write-NUnit3TestCaseAttributes {
...
+    # which is obtained by adding 0x2400
+    [int]$unicodeControlPictures = 0x2400
+    [int[]]$validChars = (0x09,0x0A,0x0D)
     $outputString = @(foreach ($o in $Output) {
         if ($null -eq $o) {
             [string]::Empty
         } else {
-            $o.ToString()
+            $result = ""
+            0..($o.Length-1) |% {
```
Comment (nohwnd) [inline src/functions/TestResults.NUnit3.ps1:551]: Could you try to replace this with regex `-replace` or similar code? This seems quite inefficient re-building the string char by char.

**Comment by ocalvo** [comment]
Hi @nohwnd , we are blocked by this, we would appreciate if you could review at your earliest convenience.

**Pull request merged**